### PR TITLE
Add CLI and checks for time-point selection

### PR DESCRIPTION
**Pull request by daniel-thom**
This extends code from @KwamiS. It is draft because some things don't work correctly. The changes in select_timepoints2.py are mostly due to me running a reformat with `black` so that whitespace and formatting are consistent.

Adds:
- CLI utility
- Checks for circuit element values before & after changing load shape values, which fails.
- Creates a partial metadata file. The existing code does not identify the condition represented by each time point.
- tests

### CODE REVIEW

File: disco/cli/select_time_points.py
Comment (daniel-thom) [inline disco/cli/select_time_points.py:83]: Here is the output of the help for this command. Please comment if you would like to change the behavior or text.

```
disco select-time-points --help
Usage: disco select-time-points [OPTIONS] MASTER_FILE

  Select load shape time points in the circuit based on the specified critical
  conditions.

  By default, the Master.dss file is not allowed enable time-series mode.
  Specify --fix-master-file to disable time-series mode and other disallowed
  parameters.
```

File: disco/preprocess/select_timepoints2.py
Comment (daniel-thom) [inline disco/preprocess/select_timepoints2.py:67]: Note that we delete any copies of the file we make in the user directory, but we do preserve a copy of what we used in the destination directory.

File: disco/preprocess/select_timepoints2.py
Comment (daniel-thom) [inline disco/preprocess/select_timepoints2.py:23]: Do we need to disallow various types of `set mode` commands? If so, what is the complete list?

File: disco/preprocess/select_timepoints2.py
Comment (daniel-thom) [inline disco/preprocess/select_timepoints2.py:244]: @KwamiS Please note this question.

File: disco/preprocess/select_timepoints2.py
Comment (daniel-thom) [inline disco/preprocess/select_timepoints2.py:307]: This is not a satisfactory comparison. There are minor comparisons that may be fixed with `math.isclose()`. Others would not. We need to resolve this before the PR can be merged.

File: disco/preprocess/select_timepoints2.py
Comment (daniel-thom) [inline disco/preprocess/select_timepoints2.py:428]: Does this feature need to be exposed to the user in the CLI? I'm not sure we even need it. The LoadShape.dss in the saved folder has all of the data, although it is in a less human-readable form.

File: disco/preprocess/select_timepoints2.py
Comment (daniel-thom) [inline disco/preprocess/select_timepoints2.py:519]: @ann-sherin Need your input on what you want here. If you prefer, you can write the code to collect what you want.

File: disco/cli/select_time_points.py
Comment (daniel-thom) [reply disco/cli/select_time_points.py:83]: My take is this:
- The user may or may not understand that we will change settings in their master file to run time-point selection. Maybe they will think that if time-series mode is enabled, we will run in that mode.
- To make sure they understand what happens, make them opt-in to these modifications by passing `--fix-master-file`.

If you all think it's OK to make all of the modifications by default, I can make those changes.

Comment (KwamiS) [reply disco/cli/select_time_points.py:83]: Looks good to me.

**Review by KwamiS — APPROVED** [review]
@daniel-thom : I think we are good to go. The only thing I thought could be modified is the num_decimals in the test code (from 2 to 3 or 4). But I don't think it's necessary.

File: disco/preprocess/select_timepoints2.py
Comment (KwamiS) [reply disco/preprocess/select_timepoints2.py:41]: Great!

**Review by KwamiS — APPROVED** [review]

File: disco/preprocess/select_timepoints2.py
Comment (KwamiS) [reply disco/preprocess/select_timepoints2.py:414]: Maybe not! I added this with the assumption that we redirect to these files. But with the "Save" and "Action=..." feature, we no longer need it.

File: disco/preprocess/select_timepoints2.py
Comment (KwamiS) [reply disco/preprocess/select_timepoints2.py:68]: Okay

File: disco/preprocess/select_timepoints2.py
Comment (KwamiS) [reply disco/preprocess/select_timepoints2.py:293]: What would be the cause of these differences?

Another TODO: We need to add the Generator object to the list of power conversion elements that can be assigned a profile, in the same category as PVSystem and Storage.

File: disco/preprocess/select_timepoints2.py
Comment (ann-sherin) [inline disco/preprocess/select_timepoints2.py:442]: I added a function to collect metadata. My thought was to get all buses and conditions associated with a specific timepoint. @KwamiS , could you review, I want to confirm that I have interpreted the outputs correctly.

also, any comments on the format?
@daniel-thom , @KwamiS

File: disco/preprocess/select_timepoints2.py
Comment (ann-sherin) [inline disco/preprocess/select_timepoints2.py:529]: should this be removed from this file?

**Comment by ann-sherin** [comment]
General comment - can we change the name of the file from select_timepoints2.py to select_critical_timepoints.py or something (mainly we can remove the suffix - 2)

File: disco/preprocess/select_timepoints2.py
Comment (ann-sherin) [inline disco/preprocess/select_timepoints2.py:498]: Is it okay to place it here? I couldnt figure out where this function is called. You can recommend another location if that is better - before it is passed to main.

Comment (daniel-thom) [reply disco/preprocess/select_timepoints2.py:498]: This works.

File: disco/preprocess/select_timepoints2.py
Comment (daniel-thom) [reply disco/preprocess/select_timepoints2.py:535]: I defer to @KwamiS on whether he wants to make users change this if they want to use it. This could be changed to use an OpenDSS feeder in the disco repo and avoid his hard-coded filesystem paths.

**Review by ann-sherin — APPROVED** [review]

**Pull request merged**
